drivers/timer/Kconfig.riscv_machine

```diff
@@ -5,7 +5,12 @@
 
 config RISCV_MACHINE_TIMER
 	bool "RISCV Machine Timer"
-	depends on SOC_FAMILY_RISCV_PRIVILEGE
+	default y
+	depends on DT_HAS_ANDESTECH_MACHINE_TIMER_ENABLED || \
+		   DT_HAS_NEORV32_MACHINE_TIMER_ENABLED || \
+		   DT_HAS_NUCLEI_SYSTIMER_ENABLED || \
+		   DT_HAS_SIFIVE_CLINT0_ENABLED || \
+		   DT_HAS_TELINK_MACHINE_TIMER_ENABLED
```

Comment:
We should probably unify these at some point, since they all enable the driver in `drivers/timer/riscv_machine_timer.c` and get rid of [this](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/timer/riscv_machine_timer.c#L12-L47) section in the driver.

Reply:
The problem is that the machine timer is not implemented the same way in all SoCs, so it's difficult (if possible?) to unify its hardware description.